develop/tutorials/articles/28-themes-and-layout-templates/xx-creating-forms-with-liferays-taglibs.markdown

```diff
@@ -0,0 +1,229 @@
+
+    <aui:fieldset >
+
+    </aui:fieldset>
+
+Alternatively, if you have multiple fieldsets, you can update your tags to be
+collapsible, using the following pattern:
+
+    <aui:fieldset collapsed="<%= true %>" collapsible="<%= true %>" >
+
+    </aui:fieldset>
+
+You can find a full list of the available attributes for the `<aui:fieldset>`
+tag in the [AUI Fieldset Taglib Docs](https://docs.liferay.com/portal/7.0/taglibs/util-taglib/aui/fieldset.html).
+
+Next, you can add your input fields.
+
+### Adding the Input Fields [](id=adding-the-input-fields)
+
+Each fieldset is used to group similar form elements together. In this section,
+you'll take a look at the different kinds of input fields you can use in your
+form.
+
+AUI input fields use the following pattern:
+
+    <aui:input label="label" name="name" type="type" />
+
+The `label` attribute sets the label for the input field. The `name` attribute
+sets the name for the field. Finally, the `type` attribute sets the type of
+input to use for the field.
+
+The `type` attribute supports the following types:
+
+- file: Adds a file browser.
+- text: The default value if no `type` is specified. Adds a text input field.
+- hidden: Adds a hidden text field.
+- assetCategories: Adds a `liferay-ui:asset-categories-selector` component.
+- assetTags: Adds a `liferay-ui:asset-tags-selector` component.
+- textarea: Adds a textarea box.
+- timeZone: Adds a `liferay-ui:input-time-zone` component.
+- password: Adds a password input field.
+- checkbox: Adds a checkbox.
+- radio: Adds a radio button.
+- color: Adds a HTML color picker.
+- editor: Adds a `liferay-ui:input-editor` component.
+- email: Adds an email input field.
+- number: Adds a number selector.
+- range: Adds a range slider.
+- resource: Adds a `liferay-ui:input-resource` component.
+- toggle-card: Adds a [Toggle Card](http://liferay.github.io/lexicon/content/toggles/#toggleCard)
+- toggle-switch: Adds a [Toggle Switch](http://liferay.github.io/lexicon/content/toggles/#toggle-switch)
+- url: Adds a URL input field.
+
+For a full list of all the attributes available for the `<aui:input>` tag, check
+out the [AUI Input Taglib Docs](https://docs.liferay.com/portal/7.0/taglibs/util-taglib/aui/input.html).
+
+Once you've added all of your input fields, you can move onto the form's buttons
+next.
+
+### Adding the Button Rows and Buttons [](id=adding-the-button-rows-and-buttons)
+
+Taking a look at the portlet configuration application example once again, you
+can see that form buttons follow the pattern below:
+
+    <aui:button-row>
+            <aui:button type="submit" />
+
+            <aui:button href="<%= redirect %>" type="cancel" />
+    </aui:button-row>
+
+The `<aui:button-row>` tags acts as a wrapper for the form's buttons, and offers
+some additional styling through the `cssClass` attribute.
+
+The `<aui:button>` tag is a standard input button, with some additional
```

Comment:
Maybe we could mention the default type is "submit" (I think) and the primary="true" option that will make that button different from other regular buttons - to emphasize it's a primary action.